cmd/devp2p/crawlercmd.go

```diff
@@ -0,0 +1,300 @@
+	}
+	return nil
+}
+
+func discv5(ctx *cli.Context, inputSet nodeSet, timeout time.Duration) nodeSet {
+	disc := startV5(ctx)
+	defer disc.Close()
+	// Crawl the DHT for some time
+	c := newCrawler(inputSet, disc, disc.RandomNodes())
+	c.revalidateInterval = 10 * time.Minute
+	return c.run(timeout)
+}
+
+func discv4(ctx *cli.Context, inputSet nodeSet, timeout time.Duration) nodeSet {
+	disc := startV4(ctx)
+	defer disc.Close()
+	// Crawl the DHT for some time
+	c := newCrawler(inputSet, disc, disc.RandomNodes())
+	c.revalidateInterval = 10 * time.Minute
+	return c.run(timeout)
+}
+
+func crawlRound(ctx *cli.Context, inputSet nodeSet, outputFile string, influxdb *influx, timeout time.Duration) nodeSet {
+	var nodes []crawledNode
+	output := make(nodeSet)
+	v5 := discv5(ctx, nodeSet{}, timeout)
+	output.add(v5.nodes()...)
+	v4 := discv4(ctx, nodeSet{}, timeout)
+	output.add(v4.nodes()...)
+	// Try to connect and get the status of all nodes
+	for _, node := range output {
+		info, err := getClientInfo(node.N)
+		if err != nil {
+			log.Warn("GetClientInfo failed", "error", err, "nodeID", node.N.ID())
+		} else {
+			log.Info("GetClientInfo succeeded")
+		}
+		nodes = append(nodes, crawledNode{node: node, info: info})
+	}
+	// Write the enodes to a file
+	writeNodesJSON(outputFile, output)
+	// Write the node info to influx
+	if influxdb != nil {
+		if err := influxdb.updateNodes(nodes); err != nil {
+			exit(err)
+		}
+	}
+	return output
+}
+
+func getClientInfo(n *enode.Node) (*clientInfo, error) {
+	var info clientInfo
+	conn, sk, err := dial(n)
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+	conn.SetDeadline(time.Now().Add(5 * time.Second))
+
+	// write hello to client
+	pub0 := crypto.FromECDSAPub(&sk.PublicKey)[1:]
+	ourHandshake := &ethtest.Hello{
+		Version: 5,
+		Caps: []p2p.Cap{
+			{Name: "eth", Version: 64},
+			{Name: "eth", Version: 65},
+			{Name: "eth", Version: 66},
+		},
+		ID: pub0,
+	}
+	if err := conn.Write(ourHandshake); err != nil {
+		return nil, err
+	}
+
+	// read hello from client
+	switch msg := conn.Read().(type) {
+	case *ethtest.Hello:
+		// set snappy if version is at least 5
+		if msg.Version >= 5 {
+			conn.SetSnappy(true)
+		}
+		info.Capabilities = msg.Caps
+		info.SoftwareVersion = msg.Version
+		info.ClientType = msg.Name
+		fmt.Printf("Caps: %v \n", msg.Caps)
+	case *ethtest.Disconnect:
+		return nil, fmt.Errorf("bad hello handshake: %v", msg.Reason.Error())
+	case *ethtest.Error:
+		return nil, fmt.Errorf("bad hello handshake: %v", msg.Error())
+	default:
+		return nil, fmt.Errorf("bad hello handshake: %v", msg.Code())
+	}
+	highestEthVersion := uint32(negotiateEthProtocol(ourHandshake.Caps, info.Capabilities))
+	// If node provides no eth version, we can skip it.
+	if highestEthVersion == 0 {
+		return &info, nil
+	}
+	conn.SetDeadline(time.Now().Add(15 * time.Second))
+	// write status message
+	status, err := getCurrentStatus()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This will make a wss-request to infura for every node you encounter, right? 
Two suggestions: cache the curent status for ~10seconds, at least, and if the call fails, use some `getDefaultStatus` which returns some hardcoded block, like 15M, and TD of whatever the current TD is and then some.